### PR TITLE
Use common format for user-agent header

### PR DESCRIPTION
**Pull request by tomblench**
## PR summary

Fixes: <! -- link to issue -->

**Note: An existing issue is [required](https://github.com/IBM/cloudant-java-sdk/blob/master/CONTRIBUTING.md#PRs) before opening a PR.**

## PR Checklist

Please make sure that your PR fulfills the following requirements:

- [x] The commit message follows the
[Angular Commit Message Guidelines](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#-commit-message-guidelines).
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type  

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] New tests
- [ ] Build/CI related changes
- [ ] Documentation content changes
- [ ] Other (please describe)

## What is the current behavior?
`user-agent` header can sometimes overflow 64 chars.

## What is the new behavior?
Use new commonly agreed format for `user-agent` header, truncating some parts where necessary.

## Does this PR introduce a breaking change?

- [] Yes
- [x] No

### CODE REVIEW

**Comment by tomblench** [comment]
NB: on my machine the user-agent looks like `cloudant-java-sdk/0.0.36(11.0.14;lin/4.18.0;Red Hat, Inc.)`

File: modules/common/src/test/java/com/ibm/cloud/cloudant/SdkCommonTest.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public void testGetSdkHeaders() {
         assertTrue(headers.containsKey("User-Agent"));
```
Comment (mojito317) [inline modules/common/src/test/java/com/ibm/cloud/cloudant/SdkCommonTest.java:37]: Can you update the copyright header to `© Copyright IBM Corporation 2020, 2022. All Rights Reserved.`?

File: modules/common/src/main/java/com/ibm/cloud/cloudant/common/SdkCommon.java [hunk 2]
```diff
@@ -16,13 +16,13 @@
 import java.io.IOException;
```
Comment (mojito317) [inline modules/common/src/main/java/com/ibm/cloud/cloudant/common/SdkCommon.java:16]: Can you update the copyright header to `© Copyright IBM Corporation 2020, 2022. All Rights Reserved.`?

Comment (tomblench) [reply modules/common/src/main/java/com/ibm/cloud/cloudant/common/SdkCommon.java]: Done

File: modules/common/src/test/java/com/ibm/cloud/cloudant/SdkCommonTest.java (hunk 1 above)
Comment (tomblench) [reply modules/common/src/test/java/com/ibm/cloud/cloudant/SdkCommonTest.java]: Now that we've gone back to the format with spaces I've reverted this file so this isn't necessary

Comment (tomblench) [reply modules/common/src/test/java/com/ibm/cloud/cloudant/SdkCommonTest.java]: Now that we've gone back to the format with spaces I've reverted this file so this isn't necessary

**Comment by mojito317** [comment]
> ~NB: on my machine the user-agent looks like `cloudant-java-sdk/0.0.36(11.0.14;lin/4.18.0;Red Hat, Inc.)`~ NB: on my machine the user-agent looks like `cloudant-java-sdk/0.0.36 (java.version=11.0.14; [java.vendor=Red Hat, Inc.]; os.name=Linux; os.version=4.18.0-348.12.2.el8_5.x86_64; os.arch=amd64; lang=java;)`

I think the `[]`s were just in the example because it is required in Java but no need for it in e.g. Node.

**Review by mojito317 — APPROVED** [review]
LGTM! 👍

**Review by eiri — APPROVED** [review]
Legit.

**Pull request merged**